File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/mainmenus/marketplace/AccountDetailsView.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus.marketplace;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.TritiumGaming.phasmophobiaevidencepicker.R;
import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.FirestoreUser;
import com.TritiumGaming.phasmophobiaevidencepicker.views.OutlineTextView;
import com.google.firebase.auth.FirebaseUser;

public class AccountDetailsView extends ConstraintLayout {

    public AccountDetailsView(@NonNull Context context) {
        super(context);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs) {
        super(context, attrs);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);

        init(getContext());
    }

    public void init(Context context) {
        inflate(context, R.layout.layout_account_details_2, this);

        FirebaseUser user = null;
        try {
            user = FirestoreUser.getCurrentFirebaseUser();
        } catch (Exception e) {
            e.printStackTrace();
        }

        setUsernameInitials(user);
        setContainerVisibility(user);
    }

    private void setContainerVisibility(FirebaseUser user) {
        if(user == null) {
            //setVisibility(GONE);
        } else {
            setVisibility(VISIBLE);
        }
    }

    private void setUsernameInitials(FirebaseUser user) {
        if(user == null) {
            return;
        }

        OutlineTextView textView_initials = findViewById(R.id.label_username_initials);
        if(textView_initials == null) {
            return;
        }

        StringBuilder displayInitials = new StringBuilder();
        String displayName = user.getDisplayName();
        if(displayName != null) {
            String[] names = displayName.split(" ");

            for(String name: names) {

                String trimmedName = name.trim();

                if(trimmedName.length() > 0) {
                    char initial = trimmedName.charAt(0);

                    displayInitials.append(initial);

                    if(displayInitials.length() >= 2) {
                        break;
                    }
                }
            }

            textView_initials.setText(displayInitials);
        }
    }

}

File: app-compose/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/ui/noncompose/views/account/AccountDetailsView.kt
package com.tritiumgaming.phasmophobiaevidencepicker.ui.views.account

import android.content.Context
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import com.google.firebase.auth.FirebaseUser
import com.tritiumgaming.phasmophobiaevidencepicker.R

class AccountDetailsView : ConstraintLayout {
    constructor(context: Context) : super(context) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
            super(context, attrs, defStyleAttr, defStyleRes) {
        init(getContext())
    }

    fun init(context: Context?) {
        inflate(context, R.layout.layout_account_details, this)

        val user = com.tritiumgaming.phasmophobiaevidencepicker.data.firebase.firestore.transactions.user.FirestoreUser.currentFirebaseUser
        setUsernameInitials(user)

        setContainerVisibility(user)
    }

    private fun setContainerVisibility(user: FirebaseUser?) {
        visibility = if (user == null) { GONE } else { VISIBLE }
    }

    @Throws(Exception::class)
    private fun setUsernameInitials(user: FirebaseUser?) {
        val accountIcon = findViewById<AccountIconView>(R.id.account_icon)
        accountIcon?.createAccountInitials(user?.displayName)
    }
}